### PR TITLE
Add moment_2.18.1

### PR DESCRIPTION
**Pull request by ns-jvanderzwaag**
@ns-ctruong @msmathers @ns-gbernsleone 

Adds latest moment version, which is required to add https://github.com/airbnb/react-dates

### CODE REVIEW

File: nodePackages/default.nix [hunk 1]
```diff
@@ -3099,6 +3099,7 @@
     module-not-found-error = callPackage ./module-not-found-error/1.0.1.nix {};
     module-not-found-error_1-0-1 = callPackage ./module-not-found-error/1.0.1.nix {};
     moment = callPackage ./moment/2.12.0.nix {};
```
Comment (ns-ctruong) [inline nodePackages/default.nix]: Any reason not to bump the default version of moment now?  Normally, it's pointed at the latest version.

**Review by ns-ctruong — COMMENTED** [review]
Just had a question.

**Comment by ns-ctruong** [comment]
Do you want to just go ahead and add react-dates in this PR too?  You'll need to do it eventually anyway.

File: nodePackages/default.nix (hunk 1 above)
Comment (ns-jvanderzwaag) [reply nodePackages/default.nix]: Good call

**Comment by ns-jvanderzwaag** [comment]
I still need to figure out how to add it since I had to fork it to add some functionality. I have a PR out to the library that will hopefully get merged and make it easy, but if that doesn't happen in time, I'll need to add it to our NPM registry or something

**Review by ns-ctruong — APPROVED** [review]
LGTM

**Pull request merged**